### ISSUE TITLE
Macros that insert access specifiers 'eat' the next DocumentSymbol

### ISSUE THREAD

**Issue by BigBahss** [microsoft/vscode-cpptools#6889]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Manjaro Linux kernel 5.10
- VS Code Version: 1.52.1-2
- C/C++ Extension Version: 1.2.0-insiders3
- Other extensions you installed (and if the issue persists after disabling them): All others disabled
- Does this issue involve using SSH remote to run the extension on a remote machine?: No
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. Create a file with the following:
```cpp
#define TEST_MACRO \
public:

class Test
{
    TEST_MACRO
// public:
public:
    Test();
    int foo();
};
```
`TEST_MACRO` can insert any access specifier.

2. Note the Outline view:
![Screenshot_2021-02-02_04-02-42](https://user-images.githubusercontent.com/58710639/106576791-89731b80-650b-11eb-95cf-1c7304ef5994.png)
The constructor `Test();` does not appear.

3. Commenting out `TEST_MACRO` and uncommenting `// public:`:
![Screenshot_2021-02-02_04-05-34](https://user-images.githubusercontent.com/58710639/106577068-ea9aef00-650b-11eb-8bd0-9c1135fd685c.png)

The next `DocumentSymbol` after the macro gets eaten. I noticed this with Qt's `Q_OBJECT` macro.

**Comment by sean-mcmanus** [comment]
That is "by design" -- you can fix by adding `#define TEST_MACRO` to a cpp.hint file in your workspace folder root.

**Comment by sean-mcmanus** [comment]
FYI, this is fixed with 1.10.3 (pre-release) -- no cpp.hint is required for this case.
